packages/editor-ui/src/components/BinaryDataDisplayEmbed.vue

```diff
@@ -18,77 +18,64 @@
 				:show-length="true"
 			/>
 			<RunDataHtml v-else-if="binaryData.fileType === 'html'" :input-html="data" />
-			<embed v-else :src="embedSource" class="binary-data" :class="embedClass()" />
+			<embed v-else :src="embedSource" class="binary-data" :class="embedClass" />
 		</span>
 	</span>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
-import { mapStores } from 'pinia';
+<script setup lang="ts">
+import { ref, onMounted, computed } from 'vue';
+import { useWorkflowsStore } from '@/stores/workflows.store';
 import type { IBinaryData } from 'n8n-workflow';
 import { jsonParse } from 'n8n-workflow';
 import type { PropType } from 'vue';
 import VueJsonPretty from 'vue-json-pretty';
-import { useWorkflowsStore } from '@/stores/workflows.store';
 import RunDataHtml from '@/components/RunDataHtml.vue';
 
-export default defineComponent({
-	name: 'BinaryDataDisplayEmbed',
-	components: {
-		VueJsonPretty,
-		RunDataHtml,
-	},
-	props: {
-		binaryData: {
-			type: Object as PropType<IBinaryData>,
-			required: true,
-		},
-	},
-	data() {
-		return {
-			isLoading: true,
-			embedSource: '',
-			error: false,
-			data: '',
-		};
-	},
-	computed: {
-		...mapStores(useWorkflowsStore),
+const props = defineProps({
```

Comment:
Can you use the `defineProps<{ ... }>()` format here?

Reply:
Yup, done